### DISCUSSION TITLE
How to template °C to climate button

### DISCUSSION THREAD

**Discussion by canedje** [matt8707/ha-fusion#350] [Q&A] [answered]
I use a tab with all my climate buttons.
That works fine but I like to template °C behind the temperature.
I tried all templates but didn't succeed.
What is the right template to achieve?
![image](https://github.com/matt8707/ha-fusion/assets/23150947/aa428141-ae96-475a-b388-4212c3010a8b)
I tried for example: {{ states.sensor.temperature.state_with_unit }} at the state tab, but it gives a wrong temperature??

![image](https://github.com/matt8707/ha-fusion/assets/23150947/8a4fa771-d2db-46ed-a450-d8156a6f1620)
(temperature is 19 C not 16.6)

**Comment by saya6k** [comment]
`{{ states(entity_id) }} °C`

**Reply by nagtegaal** [reply]
Doesn’t the following work?
`{{ state_attr(entity_id, “unit_of_measurement”) }}`

**Reply by saya6k** [reply]
<img width="649" alt="스크린샷 2024-02-11 23 01 26" src="https://github.com/matt8707/ha-fusion/assets/63517312/e330b1bb-a3c3-4a1e-ba4b-b3925ff9e08b">

This code will work.

**Comment by canedje** [comment]
both not working:
![image](https://github.com/matt8707/ha-fusion/assets/23150947/6ccea94d-97d1-470e-a8a9-ec5b985636f4)
![image](https://github.com/matt8707/ha-fusion/assets/23150947/0ea4d684-1c24-49d9-a9bc-ac3e7c1f23ec)
states(entity_id) is not representing the temp
But what is representing the temperature value?
The entity id is climate.woonkamer
There are a lot off attributes
I like to show the current_temperature
![image](https://github.com/matt8707/ha-fusion/assets/23150947/244c4444-545b-4c3b-bcb7-c5bf13e3d94f)

**Reply by saya6k** [reply]
I thought you want to use sensor. If you're using climate, there are 2 attributes, `current_temperature` and `temperature`. I don't know what you're gonna use.

**Reply by canedje** [reply]
I like to use current_temperature

**Reply by canedje** [reply, marked as answer]
I think I do have the answer:  {{ state_attr(entity_id, "current_temperature") }} °C
Thanks for the help
